mm2src/rpc_task/src/manager.rs

```diff
@@ -0,0 +1,308 @@
+use crate::{FinishedTaskResult, RpcTask, RpcTaskError, RpcTaskHandle, RpcTaskResult, RpcTaskStatus, TaskAbortHandle,
+            TaskAbortHandler, TaskId, TaskStatus, TaskStatusError, UserActionSender};
+use common::executor::spawn;
+use common::log::{debug, warn};
+use common::mm_error::prelude::*;
+use futures::channel::oneshot;
+use futures::future::{select, Either};
+use serde::Serialize;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+use std::sync::{Arc, Mutex, Weak};
+
+pub type RpcTaskManagerShared<Item, Error, InProgressStatus, AwaitingStatus, UserAction> =
+    Arc<Mutex<RpcTaskManager<Item, Error, InProgressStatus, AwaitingStatus, UserAction>>>;
+pub(crate) type RpcTaskManagerWeak<Item, Error, InProgressStatus, AwaitingStatus, UserAction> =
+    Weak<Mutex<RpcTaskManager<Item, Error, InProgressStatus, AwaitingStatus, UserAction>>>;
+
+pub struct RpcTaskManager<Item, Error, InProgressStatus, AwaitingStatus, UserAction>
+where
+    Item: Serialize,
+    Error: SerMmErrorType,
+{
+    in_progress: HashMap<TaskId, (InProgressStatus, TaskAbortHandle)>,
```

Comment:
This is what I originally tried to do. But I faced a problem when the user sends an action, and we should move `UserActionSend` and keep `TaskAbortHandle` alive somehow. Thank you, I think I found a workaround solution to use `TaskStatus` instead of separate status containers.

Reply:
Done, please check if this implementation is good enough. I haven't found a solution to avoid removing task status and returning it to the `RpcTaskManager::tasks` container.

Reply:
Yeah, it looks complicated because HashMap entries provide the API to deal with references only, but we need an owned abort_handle in this case. Maybe we can consider using Rc/Arc here, but I also find the current solution acceptable.

Reply:
Please feel free to resolve this on your behalf :slightly_smiling_face: